pkgs/applications/window-managers/dwl/default.nix

```diff
@@ -0,0 +1,79 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, pkg-config
+, libxcb
+, libxkbcommon
+, wayland
+, wayland-protocols
+, wlroots
+, enable-xwayland ? true, xwayland, libX11
+, patches ? [ ]
+, conf ? null
+, writeText
+}:
+
+stdenv.mkDerivation rec {
+  pname = "dwl";
+  version = "0.1";
+
+  src = fetchFromGitHub {
+    owner = "djpohly";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "QoKaeF5DbSX0xciwc/0VKpubn/001cJjoZ+UzVDX4qE=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [
+    libxcb
+    libxkbcommon
+    wayland
+    wayland-protocols
+    wlroots
+  ] ++ lib.optionals enable-xwayland [ xwayland libX11 ];
+
+  # Allow users to set their own list of patches
+  inherit patches;
```

Comment:
Why do we need this? This can be easily done with an overlay.

Reply:
dwl follows this strange tradition of "customize by patching the source code". Indeed they have a [dedicated page listing some patches](https://github.com/djpohly/dwl/wiki/Patches).

Reply:
I am not a fan of that practice. Not sure what the best way is here.